x-pack/metricbeat/module/googlecloud/stackdriver/metricset.go

```diff
@@ -145,3 +147,10 @@ func validatePeriodForGCP(d time.Duration) (err error) {
 
 	return nil
 }
+
+func (c *config) Validate() error {
+	if c.Region == "" && c.Zone == "" {
+		return errors.New("region and zone in Google Cloud config file cannot both be empty")
```

Comment:
should we throw an error if both region and zone are provided? What would be the expected behavior there?

Reply:
Right now the behavior is: if region is provided, then use region to get all zones under that region. If there is no region, but zone is given, then use zone. If both region and zone are provided, then region comes first.

Good point on adding an error/warning if both config parameters are given!